src/ingo_msm/msm_api.rs

```diff
@@ -317,12 +319,13 @@ impl MSMClient {
         Ok(())
     }
 
-    pub fn get_data_from_hbm(&self, data: &[u8], addr: u64, offset: u64) -> Result<Vec<u8>> {
+    pub fn get_data_from_hbm(&self, data_len: usize, addr: u64, offset: u64) -> Result<Vec<u8>> {
         log::debug!("HBM adress: {:#X?}", addr);
-        log::debug!("Data length: {:#X?}", data.len());
-        let res = self.driver_client.dma_read(addr, offset, data.len());
+        log::debug!("Data length: {:#X?}", data_len);
+        let mut res = vec![0; data_len];
```

Comment:
what if we are fetching very large data sizes, couldn't this hurt performance ?

since we are now passing data_len and allocating memory each time.

Reply:
Before it, we allocated memory inside dma_read, but in most case we had already allocated memory for reading 
moreover, we're pass reference on vector so it shouldn't hurt performance

Reply:
oh ok cool